src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/common.targets

```diff
@@ -28,31 +26,38 @@
       <ExecArgs
         Condition="'$(MvcRazorEmbedViewSources)'=='true'"
         Include="--embed-view-sources" />
+
+      <ExecArgs Include="--file=%(MvcRazorFilesToCompile.FullPath)" />
     </ItemGroup>
 
     <ItemGroup Condition="'$(SignAssembly)'=='true'">
-      <ExecArgs 
+      <ExecArgs
         Condition="'$(DelaySign)'=='true'"
         Include="--delay-sign" />
-      <ExecArgs 
+      <ExecArgs
         Condition="'$(PublicSign)'=='true'"
         Include="--public-sign" />
       <ExecArgs Include="--key-file=$(AssemblyOriginatorKeyFile)" />
     </ItemGroup>
 
     <WriteLinesToFile
       File="$(_MvcRazorResponseFilePath)"
-      Lines="@(ExecArgs)" 
+      Lines="@(ExecArgs)"
       Overwrite="true"  />
   </Target>
 
-   <Target 
-      Name="_MvcRazorPrecompileOnPublish"
-      DependsOnTargets="MvcRazorPrecompile"
-      AfterTargets="PrepareForPublish"
-      Condition="'$(MvcRazorCompileOnPublish)'=='true'">
+  <Target
+     Name="_MvcRazorPrecompileOnPublish"
+     DependsOnTargets="MvcRazorPrecompile"
+     AfterTargets="PrepareForPublish"
+     Condition="'$(MvcRazorCompileOnPublish)'=='true'">
 
     <ItemGroup>
+      <Content
+        Update="**\*.cshtml"
```

Comment:
This might still miss some cshtml files. This would be more accurate as 
`<Content Update="@(Content)" Condition="'%(Extension)'=='.cshtml'" />`.

You might also add an 'escape hatch' to the condition in case there is some reason to still publish cshtml files, even with precompliation on. Pack targets to this as a 'just in case' fallback.
`<Content Update="@(Content)" Condition="'%(Extension)'=='.cshtml' AND '%(MvcPrecompile)'!='false'" />`.